src/Microsoft.Framework.Runtime.Hosting/Json/JsonBuffer.cs

```diff
@@ -0,0 +1,301 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+
+namespace Microsoft.Framework.Runtime.Json
+{
+    internal class JsonBuffer
+    {
+        private readonly TextReader _reader;
+
+        private int _line = 1;
+        private int _column = 0;
+
+        public JsonBuffer(TextReader reader)
+        {
+            _reader = reader;
+        }
+
+        public JsonToken Read()
+        {
+            var result = new JsonToken();
+
+            int first;
+            while (true)
+            {
+                first = ReadNextChar();
+                result.Line = _line;
+                result.Column = _column;
+
+                if (first == -1)
+                {
+                    result.Type = JsonTokenType.EOL;
+                    return result;
+                }
+                else if (!IsWhitespace(first))
+                {
+                    break;
+                }
+            }
+
+            if (first == JsonConstants.LeftCurlyBracket)
+            {
+                result.Type = JsonTokenType.LeftCurlyBracket;
+            }
+            else if (first == JsonConstants.RightCurlyBracket)
+            {
+                result.Type = JsonTokenType.RightCurlyBracket;
+            }
+            else if (first == JsonConstants.LeftSquareBracket)
+            {
+                result.Type = JsonTokenType.LeftSquareBracket;
+            }
+            else if (first == JsonConstants.RightSquareBracket)
+            {
+                result.Type = JsonTokenType.RightSquareBracket;
+            }
+            else if (first == JsonConstants.Colon)
+            {
+                result.Type = JsonTokenType.Colon;
+            }
+            else if (first == JsonConstants.Comma)
+            {
+                result.Type = JsonTokenType.Comma;
+            }
+            else if (first == JsonConstants.Quotation)
+            {
+                result.Type = JsonTokenType.String;
+                result.Value = ReadString();
+            }
+            else if (first == (int)'t')
+            {
+                ReadLiteral(JsonConstants.ValueTrue);
+                result.Type = JsonTokenType.True;
+            }
+            else if (first == (int)'f')
+            {
+                ReadLiteral(JsonConstants.ValueFalse);
+                result.Type = JsonTokenType.False;
+            }
+            else if (first == (int)'n')
+            {
+                ReadLiteral(JsonConstants.ValueNull);
+                result.Type = JsonTokenType.Null;
+            }
+            else if (char.IsDigit((char)first) || first == (int)'-')
```

Comment:
`char.IsDigit` does not do what you think it does. It checks for any Unicode char, which is like hundreds of different chars. Just check for the valid ASCII range per the JSON spec.